### DISCUSSION TITLE
Firefox always shows bookmark-Sidebar after using new UI

### DISCUSSION THREAD

**Discussion by Eldon-Tyrell** [aclap-dev/video-downloadhelper#952] [Help and Q&A]
### Your question

yes - I switched to new UI - then back to the old one.
And now Firefox ALWAYS starts with the bookmark-sidebar - previously it remembered the sidebar-status on closing (bookmarks, history or disabled).
This means my Firefox is permanently damaged as I don't see a way to repair that.

I use Version 9.0.2.12

I can imagine the problem has to do with the fact that the VDH-sidebar can only be enabled WITH the new UI and switching back to the old UI is only possible via the sidebar.
If you can tell me how to fix Firefox, please do so.

another thing which has been mentioned already i think:
when using the new UI it seems impossible to use the page-title as filenames on default. 
I found the setting for it but it does nothing (is by default already on '%title" but the files have theirt original-names apparently)

this makes the whole add-on much less useful and is the reason why I don't use the new UI.

apart from that VDH works great 

### Addon & CoApp version and details

version: 9.0.2.12
target: mozilla
channel: stable
lang: de
coapp: {"found":true,"path":"C:\\Programs\\net.downloadhelper.coapp\\vdhcoapp.exe","version":"2.0.19","new_version":false}
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "used_sidebar_button": true,
    "download_directory": "E:\\Down",
    "last_download_directory": {
        "__serializer_tag": "option_some",
        "__serializer_value": {
            "__serializer_tag": "primitive",
            "__serializer_value": "E:\\Down"
        }
    },
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "use_legacy_ui": false,
    "use_sidebar": true,
    "open_count_store": 1,
    "successfull_dl": 23
}

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I am just another user.

My environment is Windows 11 Pro 64-bit version 23H2 build 22631.3810, Firefox 127.0.2 64-bit, licensed VDH 9.0.2.12a1 beta, CoApp 2.0.19.

In discussion #840 another user was having a similar problem.  Similar.  I'm not sure it's exactly the same but your problem reminds me of his.  (When you read that one, be sure to click any `Show x previous replies` & other indicators of that type.)  He ended up realizing he was just doing something wrong.  But there is also mention in there of a Firefox bug that is under investigation at Mozilla.  I don't know what the state of that problem is but you can find it from the other discussion in this forum.  It's possible that's relevant.

For your issue with the history/bookmarks sidebar, I have no help for that.  All I can say is I'm not seeing that problem.  I can open & close the browser sidebars & my Firefox does not open with either one if the sidebar was closed when I closed my last browser window.

You can switch back to the old UI from the VDH menu, not the sidebar alone:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f0c7f5c9-c494-469d-8158-99d81baad7e6)

Note the checkbox labelled `Use legacy UI` in that image of the old-style drop-down VDH menu.  That is an image of the `More settings` section of the VDH settings dialog.

Your last item about %title seems to be referring to the problem we've been discussing at length over here: #838.  (Again, click all `Show x previous replies` things you find.)  Our conclusion is that Smartnaming is simply non-functional at the moment.  It's non-functional in the VDH 9 UI AND it's non-functional in the legacy UI.  We're waiting on @paulrouget Paul to do something about it.

**Comment by Eldon-Tyrell** [comment]
GrampaWildWilly - first I thank you for the detailed answer.

I read through all of it and tried everything which was mentioned there in all imaginable variations and orders - but nothing changed.
While it may be connected to a Firefox-bug, it only started after using VDHs new UI once.

Smart-naming absolutely works for me - but only with the legacy UI.

**Comment by GrampaWildWilly** [comment]
I just added a comment to #1192 that is relevant here.

**Comment by Eldon-Tyrell** [comment]
I gave up on this and reinstalled Firefox - Then I found something on the web:

The open/closed-status of the sidebar as well as its width are only remebered, if the browser-history is set to being remembered - duh.

Extras / Settings / Security & History / History / "never remember history" --> must be unchecked

I think somebody commented this will be solved in FF-Version 133

I cannot tell if this would have solved my issue here since I already reinstalled Firefox - but it is true at the moment - i have just tested it.
